pkg/registry/apiserverinternal/storageversion/strategy.go

```diff
@@ -96,8 +103,71 @@ func (storageVersionStatusStrategy) PrepareForUpdate(ctx context.Context, obj, o
 
 	newSV.Spec = oldSV.Spec
 	metav1.ResetObjectMetaForStatus(&newSV.ObjectMeta, &oldSV.ObjectMeta)
+
+	status := apiserverinternal.ConditionFalse
+	commonEncodingVersion := "<unset>"
+	reason := commonEncodingVersionUnsetReason
+	if newSV.Status.CommonEncodingVersion != nil {
+		status = apiserverinternal.ConditionTrue
+		commonEncodingVersion = *newSV.Status.CommonEncodingVersion
+		reason = commonEncodingVersionSetReason
+	}
+	condition := apiserverinternal.StorageVersionCondition{
+		Type:   apiserverinternal.AllEncodingVersionsEqual,
+		Status: status,
+		// Generation corresponds to the desired state, which is always
+		// 0 for storage versions, because StorageVersionSpec is empty.
+		ObservedGeneration: newSV.Generation,
+		Reason:             reason,
+		Message:            fmt.Sprintf("The common encoding version is: %s", commonEncodingVersion),
+	}
```

Comment:
I think this condition should only be updated in three cases:
1. oldCommonEncodingVersion==nil, newCommonEncodingVersion!=nil,
2. oldCommonEncodingVersion!=nil, newCommonEncodingVersion==nil,
3. oldCommonEncodingVersion!=nil, newCommonEncodingVersion!=nil, and oldCommonEncodingVersion != newCommonEncodingVersion.

Reply:
I see that in setStatusCondition, you only set the lastTransition time if the condition.status has flipped, so it implemented the case 1 and 2 I mentioned above, but it doesn't include the case 3.

Instead of checking in setStatusCondition, can you do the check early on here?

Reply:
moving to a separate PR